src/bucket.js

```diff
@@ -408,10 +466,20 @@ export default class Bucket {
    * @param  {Object}   [options={}]         The options object.
    * @param  {Object}   [options.headers]    The headers object option.
    * @param  {Boolean}  [options.safe]       The safe option.
+   * @param  {Number}   [options.retry]      The retry option.
    * @param  {Boolean}  [options.aggregate]  Produces a grouped result object.
    * @return {Promise<Object, Error>}
    */
   async batch(fn, options = {}) {
-    return this.client.batch(fn, this._bucketOptions(options));
+    let batchOptions = {
+      bucket: this.name,
+      headers: this._getHeaders(options),
+      retry: this._getRetry(options),
+      safe: this._getSafe(options),
+    };
+    if (options.aggregate) {
+      batchOptions = { ...batchOptions, aggregate: options.aggregate };
+    }
```

Comment:
There are some tests using `calledWith` which (therefore) do a deep-equals equality on the resulting method arguments. See for example https://github.com/Kinto/kinto-http.js/blob/master/test/collection_test.js#L572. Unconditionally passing `aggregate` broke those tests. I didn't want to change them unnecessarily so I did this instead, although I agree it's a bit ugly. Do you think I should update the tests instead?